src/pubsub/mod_pubsub.erl

```diff
@@ -4007,50 +4004,58 @@ get_max_subscriptions_node(Host) ->
     config(serverhost(Host), max_subscriptions_node, undefined).
 
 %%%% last item cache handling
+maybe_start_cache_module(Opts) ->
+    case proplists:get_value(last_item_cache, Opts, false) of
+        false ->
+            ok;
+        Backend ->
+            start_cache_module(Backend)
+    end.
+
+start_cache_module(Backend) ->
+           gen_mod:start_backend_module(mod_pubsub_cache, [{backend, Backend}],
+                [create_table, delete_table, upsert_last_item, delete_last_item, get_last_item]),
+           mod_pubsub_cache_backend:start().
 
-is_last_item_cache_enabled({_, ServerHost, _}) ->
-    is_last_item_cache_enabled(ServerHost);
 is_last_item_cache_enabled(Host) ->
-    config(serverhost(Host), last_item_cache, false).
+    case cache_backend(Host) of
+        false ->
+            false;
+        _ ->
+            true
+    end.
+
+cache_backend(Host) ->
+     gen_mod:get_module_opt(serverhost(Host), mod_pubsub, last_item_cache, false).
 
-set_cached_item({_, ServerHost, _}, Nidx, ItemId, Publisher, Payload) ->
-    set_cached_item(ServerHost, Nidx, ItemId, Publisher, Payload);
+set_cached_item({_, Host, _}, Nidx, ItemId, Publisher, Payload) ->
+    set_cached_item(Host, Nidx, ItemId, Publisher, Payload);
 set_cached_item(Host, Nidx, ItemId, Publisher, Payload) ->
-    case is_last_item_cache_enabled(Host) of
-        true -> mnesia:dirty_write({pubsub_last_item, Nidx, ItemId,
-                                    {timestamp(), jid:to_lower(jid:to_bare(Publisher))},
-                                    Payload});
-        _ -> ok
-    end.
+    is_last_item_cache_enabled(Host) andalso
+        mod_pubsub_cache_backend:upsert_last_item(Nidx, ItemId, Publisher, Payload).
 
-unset_cached_item({_, ServerHost, _}, Nidx) ->
-    unset_cached_item(ServerHost, Nidx);
+unset_cached_item({_, Host, _}, Nidx) ->
+    unset_cached_item(Host, Nidx);
 unset_cached_item(Host, Nidx) ->
-    case is_last_item_cache_enabled(Host) of
-        true -> mnesia:dirty_delete({pubsub_last_item, Nidx});
-        _ -> ok
-    end.
+    is_last_item_cache_enabled(Host) andalso
+        mod_pubsub_cache_backend:delete_last_item(serverhost(Host), Nidx).
 
 -spec get_cached_item(
         Host    :: mod_pubsub:host(),
           Nidx :: mod_pubsub:nodeIdx())
-        -> undefined | mod_pubsub:pubsubItem().
-get_cached_item({_, ServerHost, _}, Nidx) ->
-    get_cached_item(ServerHost, Nidx);
+        -> false | mod_pubsub:pubsubItem().
+get_cached_item({_, Host, _}, Nidx) ->
```

Comment:
I have a hunch this line was actually correct and applied to PEP.

Reply:
Actually, if out Serverhost is `<<"localhost">>` the argument which is passed to this function is `<<"pubsub.localhost">>`